x-pack/plugin/security/cli/src/main/java/org/elasticsearch/xpack/security/cli/CertificateTool.java

```diff
@@ -807,7 +819,8 @@ void generateAndWriteSignedCertificates(Path output, boolean writeZipFile, Optio
                         } else {
                             final String fileName = entryBase + ".p12";
                             outputStream.putNextEntry(new ZipEntry(fileName));
-                            writePkcs12(fileName, outputStream, certificateInformation.name.originalName, pair, caInfo.certAndKey.cert,
+                            writePkcs12(fileName, outputStream, certificateInformation.name.originalName, pair,
```

Comment:
I also _think_ we don't need to add a certificate entry too when we create a self-signed PKCS12, but did we check?

Reply:
This is a good point and it turns out the check is missing ...
I now [augumented](https://github.com/elastic/elasticsearch/pull/64037/files#diff-f63090ebeefa52865c05fb313b5d7e73fd81a4c37f338a7a328df9c74b6134b7R634) `testCreateCaAndMultipleInstances` to test p12 generation for self-signed cert.

Reply:
My point was actually about the construction of the PKCS#12 bundle and whether or not it suffices to add a private key entry with the key and certificate, or if we also need to add a trusted certificate entry with the certificate too. I believe we don't and other tooling ( keytool, openssl ) seem to agree too :)